### ISSUE TITLE
Unable to build exa on osx 10.13.2

### ISSUE THREAD

**Issue by varunpatro** [ogham/exa#353]
Hello I'm having difficulties building `exa` on my mac.

Here's the full build log:

```
[nix-shell:~/workshop/exa]$ cargo build -v
       Fresh winapi-build v0.1.1
       Fresh byteorder v0.4.2
       Fresh utf8-ranges v0.1.3
       Fresh pkg-config v0.3.9
       Fresh matches v0.1.6
       Fresh winapi v0.2.8
       Fresh glob v0.2.11
       Fresh unicode-width v0.1.4
       Fresh libc v0.2.30
       Fresh log v0.3.8
       Fresh ansi_term v0.10.2
       Fresh unicode-normalization v0.1.5
       Fresh lazy_static v0.2.8
       Fresh num-traits v0.1.40
       Fresh percent-encoding v1.0.0
       Fresh regex-syntax v0.3.9
       Fresh natord v1.0.9
       Fresh nom v1.2.4
       Fresh scoped_threadpool v0.1.7
       Fresh gcc v0.3.53
       Fresh bitflags v0.9.1
       Fresh unicode-bidi v0.3.4
       Fresh term_grid v0.1.6
       Fresh pad v0.1.4
       Fresh num_cpus v1.6.2
       Fresh users v0.5.3
       Fresh locale v0.2.2
       Fresh memchr v0.1.11
       Fresh term_size v0.3.0
       Fresh number_prefix v0.2.7
       Fresh iso8601 v0.1.1
       Fresh cmake v0.1.25
       Fresh idna v0.1.4
       Fresh aho-corasick v0.5.3
       Fresh datetime v0.4.7
       Fresh kernel32-sys v0.2.2
       Fresh url v1.5.1
       Fresh zoneinfo_compiled v0.4.7
       Fresh libz-sys v1.0.16
       Fresh thread-id v2.0.0
       Fresh libgit2-sys v0.6.14
       Fresh thread_local v0.2.7
       Fresh git2 v0.6.8
       Fresh regex v0.1.80
       Fresh env_logger v0.3.5
   Compiling exa v0.9.0-pre (file:///Users/varun/workshop/exa)
     Running `rustc --crate-name exa src/bin/main.rs --crate-type bin --emit=dep-info,link -C debuginfo=2 --cfg 'feature="default"' --cfg 'feature="git"' --cfg 'feature="git2"' -C metadata=0cd0cfd62d081e3d -C extra-filename=-0cd0cfd62d081e3d --out-dir /Users/varun/workshop/exa/target/debug/deps -L dependency=/Users/varun/workshop/exa/target/debug/deps --extern zoneinfo_compiled=/Users/varun/workshop/exa/target/debug/deps/libzoneinfo_compiled-56b83576ef95f125.rlib --extern locale=/Users/varun/workshop/exa/target/debug/deps/liblocale-f9c4b7395a8e850f.rlib --extern env_logger=/Users/varun/workshop/exa/target/debug/deps/libenv_logger-0c495ad83bb4b711.rlib --extern git2=/Users/varun/workshop/exa/target/debug/deps/libgit2-cbdc010f6df824bd.rlib --extern term_size=/Users/varun/workshop/exa/target/debug/deps/libterm_size-1973b5fff6d5e511.rlib --extern scoped_threadpool=/Users/varun/workshop/exa/target/debug/deps/libscoped_threadpool-a08a541ad792202d.rlib --extern datetime=/Users/varun/workshop/exa/target/debug/deps/libdatetime-5bc9d97e99acfb20.rlib --extern ansi_term=/Users/varun/workshop/exa/target/debug/deps/libansi_term-0103310ea5603814.rlib --extern libc=/Users/varun/workshop/exa/target/debug/deps/liblibc-c24881d342fd125f.rlib --extern unicode_width=/Users/varun/workshop/exa/target/debug/deps/libunicode_width-68bacc3856408965.rlib --extern num_cpus=/Users/varun/workshop/exa/target/debug/deps/libnum_cpus-2beb4fa4ccd7f742.rlib --extern glob=/Users/varun/workshop/exa/target/debug/deps/libglob-5d893a44405f91ce.rlib --extern log=/Users/varun/workshop/exa/target/debug/deps/liblog-0a02255918485fbf.rlib --extern users=/Users/varun/workshop/exa/target/debug/deps/libusers-b8b8e6b8c5de23fa.rlib --extern natord=/Users/varun/workshop/exa/target/debug/deps/libnatord-8b4f92d595411d28.rlib --extern lazy_static=/Users/varun/workshop/exa/target/debug/deps/liblazy_static-707de6d4492cf977.rlib --extern number_prefix=/Users/varun/workshop/exa/target/debug/deps/libnumber_prefix-a445b2dea6e1522e.rlib --extern term_grid=/Users/varun/workshop/exa/target/debug/deps/libterm_grid-89fa02262f8c298b.rlib --extern exa=/Users/varun/workshop/exa/target/debug/deps/libexa-169ddbcc1901dc9c.rlib -L native=/Users/varun/workshop/exa/target/debug/build/libgit2-sys-785c8d19657e32b3/out/lib`
error: linking with `/nix/store/2fai9mvfi9nak6jxmqnvv9xqlq0qlinb-clang-wrapper-4.0.1/bin/cc` failed: exit code: 1
  |
  = note: "/nix/store/2fai9mvfi9nak6jxmqnvv9xqlq0qlinb-clang-wrapper-4.0.1/bin/cc" "-m64" "-L" "/nix/store/raj25sxp1bi4qky5kj4cq0yd06lp8bid-rustc-1.22.1/lib/rustlib/x86_64-apple-darwin/lib" "/Users/varun/workshop/exa/target/debug/deps/exa-0cd0cfd62d081e3d.exa0.rust-cgu.o" "/Users/varun/workshop/exa/target/debug/deps/exa-0cd0cfd62d081e3d.exa1.rust-cgu.o" "/Users/varun/workshop/exa/target/debug/deps/exa-0cd0cfd62d081e3d.exa10.rust-cgu.o" "/Users/varun/workshop/exa/target/debug/deps/exa-0cd0cfd62d081e3d.exa11.rust-cgu.o" "/Users/varun/workshop/exa/target/debug/deps/exa-0cd0cfd62d081e3d.exa12.rust-cgu.o" "/Users/varun/workshop/exa/target/debug/deps/exa-0cd0cfd62d081e3d.exa13.rust-cgu.o" "/Users/varun/workshop/exa/target/debug/deps/exa-0cd0cfd62d081e3d.exa14.rust-cgu.o" "/Users/varun/workshop/exa/target/debug/deps/exa-0cd0cfd62d081e3d.exa15.rust-cgu.o" "/Users/varun/workshop/exa/target/debug/deps/exa-0cd0cfd62d081e3d.exa2.rust-cgu.o" "/Users/varun/workshop/exa/target/debug/deps/exa-0cd0cfd62d081e3d.exa3.rust-cgu.o" "/Users/varun/workshop/exa/target/debug/deps/exa-0cd0cfd62d081e3d.exa4.rust-cgu.o" "/Users/varun/workshop/exa/target/debug/deps/exa-0cd0cfd62d081e3d.exa5.rust-cgu.o" "/Users/varun/workshop/exa/target/debug/deps/exa-0cd0cfd62d081e3d.exa6.rust-cgu.o" "/Users/varun/workshop/exa/target/debug/deps/exa-0cd0cfd62d081e3d.exa7.rust-cgu.o" "/Users/varun/workshop/exa/target/debug/deps/exa-0cd0cfd62d081e3d.exa8.rust-cgu.o" "/Users/varun/workshop/exa/target/debug/deps/exa-0cd0cfd62d081e3d.exa9.rust-cgu.o" "-o" "/Users/varun/workshop/exa/target/debug/deps/exa-0cd0cfd62d081e3d" "/Users/varun/workshop/exa/target/debug/deps/exa-0cd0cfd62d081e3d.crate.allocator.rust-cgu.o" "-Wl,-dead_strip" "-nodefaultlibs" "-L" "/Users/varun/workshop/exa/target/debug/deps" "-L" "/Users/varun/workshop/exa/target/debug/build/libgit2-sys-785c8d19657e32b3/out/lib" "-L" "/nix/store/raj25sxp1bi4qky5kj4cq0yd06lp8bid-rustc-1.22.1/lib/rustlib/x86_64-apple-darwin/lib" "/Users/varun/workshop/exa/target/debug/deps/libexa-169ddbcc1901dc9c.rlib" "/Users/varun/workshop/exa/target/debug/deps/libnum_cpus-2beb4fa4ccd7f742.rlib" "/Users/varun/workshop/exa/target/debug/deps/libterm_grid-89fa02262f8c298b.rlib" "/Users/varun/workshop/exa/target/debug/deps/liblazy_static-707de6d4492cf977.rlib" "/Users/varun/workshop/exa/target/debug/deps/libnatord-8b4f92d595411d28.rlib" "/Users/varun/workshop/exa/target/debug/deps/libscoped_threadpool-a08a541ad792202d.rlib" "/Users/varun/workshop/exa/target/debug/deps/libzoneinfo_compiled-56b83576ef95f125.rlib" "/Users/varun/workshop/exa/target/debug/deps/libdatetime-5bc9d97e99acfb20.rlib" "/Users/varun/workshop/exa/target/debug/deps/libiso8601-c928953a06da9819.rlib" "/Users/varun/workshop/exa/target/debug/deps/libnom-b0376c5627353bb8.rlib" "/Users/varun/workshop/exa/target/debug/deps/libpad-8f3c915f15f1a9cf.rlib" "/Users/varun/workshop/exa/target/debug/deps/liblocale-f9c4b7395a8e850f.rlib" "/Users/varun/workshop/exa/target/debug/deps/libbyteorder-aa3d5eadc1df00a0.rlib" "/Users/varun/workshop/exa/target/debug/deps/libgit2-cbdc010f6df824bd.rlib" "/Users/varun/workshop/exa/target/debug/deps/liblibgit2_sys-657c768d4d4031a9.rlib" "/Users/varun/workshop/exa/target/debug/deps/liburl-b7fdafb93ad9ee34.rlib" "/Users/varun/workshop/exa/target/debug/deps/libidna-547b542a037eac01.rlib" "/Users/varun/workshop/exa/target/debug/deps/libunicode_bidi-7626d81e0c190f3c.rlib" "/Users/varun/workshop/exa/target/debug/deps/libpercent_encoding-d73d7db07777d975.rlib" "/Users/varun/workshop/exa/target/debug/deps/libbitflags-11d0118d0906c817.rlib" "/Users/varun/workshop/exa/target/debug/deps/libenv_logger-0c495ad83bb4b711.rlib" "/Users/varun/workshop/exa/target/debug/deps/libregex-a7d6e9c435e052b9.rlib" "/Users/varun/workshop/exa/target/debug/deps/libutf8_ranges-52ac9ee3e477f24e.rlib" "/Users/varun/workshop/exa/target/debug/deps/libthread_local-e7feb1b4dd0d9b05.rlib" "/Users/varun/workshop/exa/target/debug/deps/libregex_syntax-70183370697aeda7.rlib" "/Users/varun/workshop/exa/target/debug/deps/libansi_term-0103310ea5603814.rlib" "/Users/varun/workshop/exa/target/debug/deps/libunicode_width-68bacc3856408965.rlib" "/Users/varun/workshop/exa/target/debug/deps/libglob-5d893a44405f91ce.rlib" "/Users/varun/workshop/exa/target/debug/deps/libnumber_prefix-a445b2dea6e1522e.rlib" "/Users/varun/workshop/exa/target/debug/deps/libnum_traits-e4f0d0c7324b5373.rlib" "/Users/varun/workshop/exa/target/debug/deps/libusers-b8b8e6b8c5de23fa.rlib" "/Users/varun/workshop/exa/target/debug/deps/libterm_size-1973b5fff6d5e511.rlib" "/Users/varun/workshop/exa/target/debug/deps/libmatches-18f5d6678e66b12b.rlib" "/Users/varun/workshop/exa/target/debug/deps/libunicode_normalization-13a55b3fdd44286c.rlib" "/Users/varun/workshop/exa/target/debug/deps/liblibz_sys-f8c5023ae7c4c289.rlib" "/Users/varun/workshop/exa/target/debug/deps/liblog-0a02255918485fbf.rlib" "/Users/varun/workshop/exa/target/debug/deps/libaho_corasick-c2cf9aedc349206f.rlib" "/Users/varun/workshop/exa/target/debug/deps/libmemchr-e1fa6a3cc7ea4327.rlib" "/Users/varun/workshop/exa/target/debug/deps/libthread_id-4c7b6d3c00ee649a.rlib" "/Users/varun/workshop/exa/target/debug/deps/liblibc-c24881d342fd125f.rlib" "/nix/store/raj25sxp1bi4qky5kj4cq0yd06lp8bid-rustc-1.22.1/lib/rustlib/x86_64-apple-darwin/lib/libstd-a5573c28baa2d270.rlib" "/nix/store/raj25sxp1bi4qky5kj4cq0yd06lp8bid-rustc-1.22.1/lib/rustlib/x86_64-apple-darwin/lib/libpanic_unwind-0e1dd264eb0295ca.rlib" "/nix/store/raj25sxp1bi4qky5kj4cq0yd06lp8bid-rustc-1.22.1/lib/rustlib/x86_64-apple-darwin/lib/liballoc_jemalloc-40e38640b113cbe4.rlib" "/nix/store/raj25sxp1bi4qky5kj4cq0yd06lp8bid-rustc-1.22.1/lib/rustlib/x86_64-apple-darwin/lib/libunwind-4d7cc397a07cb119.rlib" "/nix/store/raj25sxp1bi4qky5kj4cq0yd06lp8bid-rustc-1.22.1/lib/rustlib/x86_64-apple-darwin/lib/liballoc_system-3555cc0ff631180a.rlib" "/nix/store/raj25sxp1bi4qky5kj4cq0yd06lp8bid-rustc-1.22.1/lib/rustlib/x86_64-apple-darwin/lib/liblibc-d3676d01f6a9d053.rlib" "/nix/store/raj25sxp1bi4qky5kj4cq0yd06lp8bid-rustc-1.22.1/lib/rustlib/x86_64-apple-darwin/lib/liballoc-0285aed88b3edf43.rlib" "/nix/store/raj25sxp1bi4qky5kj4cq0yd06lp8bid-rustc-1.22.1/lib/rustlib/x86_64-apple-darwin/lib/libstd_unicode-8a7521ee3925d10e.rlib" "/nix/store/raj25sxp1bi4qky5kj4cq0yd06lp8bid-rustc-1.22.1/lib/rustlib/x86_64-apple-darwin/lib/librand-7270a913c138d53b.rlib" "/nix/store/raj25sxp1bi4qky5kj4cq0yd06lp8bid-rustc-1.22.1/lib/rustlib/x86_64-apple-darwin/lib/libcore-24fa1060c4e0b7b1.rlib" "/nix/store/raj25sxp1bi4qky5kj4cq0yd06lp8bid-rustc-1.22.1/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-eeb63a38b1651c16.rlib" "-l" "iconv" "-framework" "Security" "-framework" "CoreFoundation" "-l" "z" "-l" "System" "-l" "resolv" "-l" "pthread" "-l" "c" "-l" "m"
  = note: ld: library not found for -liconv
          clang-4.0: error: linker command failed with exit code 1 (use -v to see invocation)


error: aborting due to previous error

error: Could not compile `exa`.

Caused by:
  process didn't exit successfully: `rustc --crate-name exa src/bin/main.rs --crate-type bin --emit=dep-info,link -C debuginfo=2 --cfg feature="default" --cfg feature="git" --cfg feature="git2" -C metadata=0cd0cfd62d081e3d -C extra-filename=-0cd0cfd62d081e3d --out-dir /Users/varun/workshop/exa/target/debug/deps -L dependency=/Users/varun/workshop/exa/target/debug/deps --extern zoneinfo_compiled=/Users/varun/workshop/exa/target/debug/deps/libzoneinfo_compiled-56b83576ef95f125.rlib --extern locale=/Users/varun/workshop/exa/target/debug/deps/liblocale-f9c4b7395a8e850f.rlib --extern env_logger=/Users/varun/workshop/exa/target/debug/deps/libenv_logger-0c495ad83bb4b711.rlib --extern git2=/Users/varun/workshop/exa/target/debug/deps/libgit2-cbdc010f6df824bd.rlib --extern term_size=/Users/varun/workshop/exa/target/debug/deps/libterm_size-1973b5fff6d5e511.rlib --extern scoped_threadpool=/Users/varun/workshop/exa/target/debug/deps/libscoped_threadpool-a08a541ad792202d.rlib --extern datetime=/Users/varun/workshop/exa/target/debug/deps/libdatetime-5bc9d97e99acfb20.rlib --extern ansi_term=/Users/varun/workshop/exa/target/debug/deps/libansi_term-0103310ea5603814.rlib --extern libc=/Users/varun/workshop/exa/target/debug/deps/liblibc-c24881d342fd125f.rlib --extern unicode_width=/Users/varun/workshop/exa/target/debug/deps/libunicode_width-68bacc3856408965.rlib --extern num_cpus=/Users/varun/workshop/exa/target/debug/deps/libnum_cpus-2beb4fa4ccd7f742.rlib --extern glob=/Users/varun/workshop/exa/target/debug/deps/libglob-5d893a44405f91ce.rlib --extern log=/Users/varun/workshop/exa/target/debug/deps/liblog-0a02255918485fbf.rlib --extern users=/Users/varun/workshop/exa/target/debug/deps/libusers-b8b8e6b8c5de23fa.rlib --extern natord=/Users/varun/workshop/exa/target/debug/deps/libnatord-8b4f92d595411d28.rlib --extern lazy_static=/Users/varun/workshop/exa/target/debug/deps/liblazy_static-707de6d4492cf977.rlib --extern number_prefix=/Users/varun/workshop/exa/target/debug/deps/libnumber_prefix-a445b2dea6e1522e.rlib --extern term_grid=/Users/varun/workshop/exa/target/debug/deps/libterm_grid-89fa02262f8c298b.rlib --extern exa=/Users/varun/workshop/exa/target/debug/deps/libexa-169ddbcc1901dc9c.rlib -L native=/Users/varun/workshop/exa/target/debug/build/libgit2-sys-785c8d19657e32b3/out/lib` (exit code: 101)
```

**Comment by kballard** [comment]
I'm hitting this too. It affects manual builds as well, not just builds from nix. The full error with the manual build is

```
error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-m64" "-L" "/Users/kevin/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" "/Users/kevin/Dev/rust/exa/target/release/deps/exa-b4615fc1c17dc3db.exa9.rcgu.o" "-o" "/Users/kevin/Dev/rust/exa/target/release/deps/exa-b4615fc1c17dc3db" "-Wl,-dead_strip" "-nodefaultlibs" "-L" "/Users/kevin/Dev/rust/exa/target/release/deps" "-L" "/Users/kevin/Dev/rust/exa/target/release/build/libgit2-sys-d6416c580a74d9b5/out/lib" "-L" "/Users/kevin/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" "/var/folders/cs/l6c6g06x0974_0w5pdg8dps80000gn/T/rustc.CP5mTF4IUMVe/liblibgit2_sys-230998e8f1e0b4eb.rlib" "/var/folders/cs/l6c6g06x0974_0w5pdg8dps80000gn/T/rustc.CP5mTF4IUMVe/libstd-ab5ab2cb870d2bce.rlib" "/var/folders/cs/l6c6g06x0974_0w5pdg8dps80000gn/T/rustc.CP5mTF4IUMVe/liballoc_jemalloc-a84d38b17837bc6a.rlib" "/Users/kevin/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-4fea139ca33d24bf.rlib" "-l" "iconv" "-framework" "Security" "-framework" "CoreFoundation" "-l" "z" "-l" "System" "-l" "resolv" "-l" "pthread" "-l" "c" "-l" "m"
  = note: Undefined symbols for architecture x86_64:
            "_libiconv", referenced from:
                _git_path_iconv in liblibgit2_sys-230998e8f1e0b4eb.rlib(path.c.o)
            "_libiconv_open", referenced from:
                _git_path_direach in liblibgit2_sys-230998e8f1e0b4eb.rlib(path.c.o)
                _git_path_iconv_init_precompose in liblibgit2_sys-230998e8f1e0b4eb.rlib(path.c.o)
                _git_path_diriter_init in liblibgit2_sys-230998e8f1e0b4eb.rlib(path.c.o)
            "_libiconv_close", referenced from:
                _git_path_direach in liblibgit2_sys-230998e8f1e0b4eb.rlib(path.c.o)
                _git_path_iconv_clear in liblibgit2_sys-230998e8f1e0b4eb.rlib(path.c.o)
                _git_path_diriter_free in liblibgit2_sys-230998e8f1e0b4eb.rlib(path.c.o)
          ld: symbol(s) not found for architecture x86_64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by kballard** [comment]
Disabling the `git` feature works around this issue (though obviously at the cost of disabling all git-related functionality).

**Comment by kballard** [comment]
https://github.com/alexcrichton/git2-rs/issues/263 is the underlying issue in git2-rs.

**Comment by ariasuni** [comment]
The git2-rs issue has been fixed and exa uses a recent version since 8046980806841a10571a497bcdc8dbf1c3c08c7a. Can you confirm this is fixed?

**Comment by ariasuni** [comment]
I’ve successfully compiled latest exa with latest Rust on macOS Big Sur, so I take that as a confirmation that it now works.
